### PR TITLE
sync lvgl/lv_fs_if

### PR DESCRIPTION
**Pull request by LikWoody**
### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Checkpoints
- [x] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Update [CHANGELOG.md](https://github.com/lvgl/lvgl/blob/master/docs/CHANGELOG.md)
- [ ] Update the documentation

### CODE REVIEW

File: src/extra/libs/fs/lv_fs_fatfs.c [hunk 1]
```diff
@@ -252,7 +267,9 @@ static lv_fs_res_t fs_dir_read (lv_fs_drv_t * drv, void * dir_p, char *fn)
 static lv_fs_res_t fs_dir_close (lv_fs_drv_t * drv, void * dir_p)
 {
 	f_closedir(dir_p);
+    lv_mem_free(dir_p);
     return LV_FS_RES_OK;
 }
 
-#endif  /*LV_USE_FS_FATFS*/
+#endif	/*LV_USE_FS_IF*/
+#endif  /*LV_FS_IF_FATFS*/
```
Comment (kisvegabor) [inline src/extra/libs/fs/lv_fs_fatfs.c]: Same here.

File: src/extra/libs/fs/lv_fs_fatfs.c [hunk 2]
```diff
@@ -6,9 +6,10 @@
 /*********************
  *      INCLUDES
  *********************/
-#include "../../../lvgl.h"
+#include "lv_fs_if.h"
 
-#if LV_USE_FS_FATFS != '\0'
+#if LV_USE_FS_IF
+#if LV_FS_IF_FATFS != '\0'
```
Comment (kisvegabor) [inline src/extra/libs/fs/lv_fs_fatfs.c]: There should be only `LV_USE_FS_FATFS` here.

File: src/extra/libs/fs/lv_fs_fatfs.c [hunk 3]
```diff
@@ -62,7 +63,7 @@ void lv_fs_if_fatfs_init(void)
 	lv_fs_drv_init(&fs_drv);
 
     /*Set up fields...*/
-    fs_drv.letter = LV_USE_FS_FATFS;
+    fs_drv.letter = LV_FS_IF_FATFS;
```
Comment (kisvegabor) [inline src/extra/libs/fs/lv_fs_fatfs.c]: It called `LV_USE_FS_FATFS` here.

**Review by kisvegabor — COMMENTED** [review]
Thank you.

I found a few copy paste issues.

**Comment by LikWoody** [comment]
sorry for my carelessness.

File: src/extra/libs/fs/lv_fs_fatfs.c [hunk 4]
```diff
@@ -6,7 +6,7 @@
 /*********************
  *      INCLUDES
  *********************/
-#include "../../../lvgl.h"
+#include "lv_fs_if.h"
```
Comment (kisvegabor) [inline src/extra/libs/fs/lv_fs_fatfs.c]:
```suggestion
#include "../../../lvgl.h"
```

**Comment by kisvegabor** [comment]
Thank you, merging.

**Pull request merged**